### DISCUSSION TITLE
Restrict Access to Certain Secrets by User

### DISCUSSION THREAD

**Discussion by jonathan-boudreau-work** [community/community#20396] [Enterprise]
Is there a way to prevent someone who has access to a repository from reading production secrets? We don't want to allow all of our developers to access production servers unless needed. The issue is there doesn't seem to be a way to restrict deployment secrets with github actions. You either have access to all secrets on the repository or none at all.

As an example, CircleCI allows restricting context access by who triggered the build: [https://circleci.com/blog/protect-secrets-with-restricted-contexts/](https://circleci.com/blog/protect-secrets-with-restricted-contexts/)

**Comment by jsoref** [comment]
You might consider an environment, although they're apparently only generally available for public repositories (although they are available to private repositories in GitHub Enterprise Cloud):
https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment

You could do magical things where you have a second repository which is owned by a different entity (if you're using an enterprise, that could be another org, I guess) and have some automation that crosses the PR over. But it's worth keeping mind that if someone is malicious, they'll figure out how to attack your system (even through airgaps), so your goal should be to have trustworthy people... -- functionally, [github/docs](https://github.com/github/docs) is related to another github repository which isn't public and uses all sorts of automation (although possibly not this precise kind).

**Reply by jonathan-boudreau-work** [reply]
Sounds like we'll be sticking with CircleCi then. Thanks anyways.
